src/overlays/package_dump.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs::File;
+use std::io::Write;
+use destiny_pkg::TagHash;
+use imgui::Ui;
+use winit::window::Window;
+use crate::overlays::gui::OverlayProvider;
+use crate::packages::package_manager;
+use crate::resources::Resources;
+
+
+pub struct PackageDumper {
+    package_id: String,
+    entry_id: String,
+    message: Result<String, String>
+}
+
+impl PackageDumper {
+
+    pub fn new() -> PackageDumper {
+        return PackageDumper{ package_id: "".to_string(), entry_id: "".to_string(), message: Ok("".to_string()) };
+    }
+
+    fn dump_entry(&self, pkg_id: u16, entry_id: u16) -> bool {
+        let tag = TagHash::new(pkg_id, entry_id);
+        let entry_header = package_manager().get_entry(tag);
+        if entry_header.is_ok() {
+            let entry = entry_header.unwrap();
+            let mut name = format!("{:#010X}", entry.reference);
```

Comment:
Why 10 digits for an 8-digit hexadecimal?

Reply:
Already fixed that one, but according to an online thing, the 0x is included in the length when formatting as hex.

Reply:
Ah that's funky, didn't know that the "prettify" marker (#) worked with other types, so thats another lesson for me